packages/template-retail-react-app/app/utils/url.js

```diff
@@ -149,23 +150,95 @@ export const getPathWithLocale = (shortCode, opts = {}) => {
 
     const isDefaultLocaleOfDefaultSite = shortCode === defaultSite.l10n.defaultLocale
     const isDefaultSite = siteRef === defaultSite.alias || siteRef === defaultSite.id
+
+    // Remove query parameters
+    const {disallowParams = []} = opts
+
+    let queryString = new URLSearchParams(`${search}`)
+
+    if (disallowParams.length) {
+        disallowParams.forEach((param) => {
+            queryString.delete(param)
+        })
+    }
+
     // rebuild the url with new locale,
-    const newUrl = buildPathWithUrlConfig(
-        `${pathname}${search}`,
-        {
-            // By default, as for home page, when the values of site and locale belongs to the default site,
-            // they will be not shown in the url just
-            site:
-                isDefaultLocaleOfDefaultSite && isDefaultSite && isHomeRef
-                    ? ''
-                    : siteRef || defaultSite.alias || defaultSite.id,
-            locale: isDefaultLocaleOfDefaultSite && isDefaultSite && isHomeRef ? '' : shortCode
-        },
-        opts
+    const newUrl = fillUrlTemplate(
+        `${pathname}${Array.from(queryString).length !== 0 ? `?${queryString}` : ''}`,
+        // By default, as for home page, when the values of site and locale belongs to the default site,
+        // they will be not shown in the url just
+        isDefaultLocaleOfDefaultSite && isDefaultSite && isHomeRef
+            ? ''
+            : siteRef || defaultSite.alias || defaultSite.id,
+        isDefaultLocaleOfDefaultSite && isDefaultSite && isHomeRef ? '' : shortCode
```

Comment:
We've seen this logic before. Shouldn't the `createUrlTemplate` handle whether or not defaults are shown or not? That function to my knowledge as access to the site or locale and the config that will tell it if the site or locale is the default and whether or not you want to show defaults?